app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt

```diff
@@ -24,6 +25,12 @@ class FiltersActivity : BaseActivity(), FiltersListener {
     private val binding by viewBinding(ActivityFiltersBinding::inflate)
     private val viewModel: FiltersViewModel by viewModels()
 
+    private val editFilterLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            // refresh the filters upon returning from EditFilterActivity
+            loadFilters()
```

Comment:
should we perhaps force load here, even if previous one is in progress?

Reply:
ah I guess it's addressed in the next commit already?

Reply:
Yes, I changed `loadFilters()` to `reloadFilters()`.
Now filters are loaded automatically when the ViewModel is created and reloaded when `reloadFilters()` is called, cancelling the in-progress load if any. This is done using the `collectLatest()` function which cancels the processing of the previous item when a new one arrives.